ynr/apps/cached_counts/templates/reports.html

```diff
@@ -50,17 +77,17 @@ <h4>Statistics for {{ election.election_name }} on {{ election.election_date }}<
   <div class="pagination">
     <span class="step-links">
         {% if page_obj.has_previous %}
-          <a href="?page=1&q={{ query }}">&laquo; first</a>
-          <a href="?page={{ page_obj.previous_page_number }}&q={{ query }}">previous</a>
+          <a href="?page=1&{{ filter_params }}">&laquo; first</a>
```

Comment:
You might want to look at https://github.com/DemocracyClub/yournextrepresentative/blob/76eea0c06a5e53f77d4ffd07da94ac83fe3905c9/ynr/apps/uk_results/templatetags/get_query.py#L11.

Django 5.1+ has this built-in: https://docs.djangoproject.com/en/5.2/ref/templates/builtins/#querystring